### PR TITLE
Add deploy_kedro_viz job to CircleCI

### PR DESCRIPTION
**Pull request by tynandebold**
## Description
In Kedro-Viz, we've sometimes seen errors in our builds after a change made to Kedro. This PR adds a build step that will kick of a Kedro-Viz CircleCI build each time there's a change to the master branch here in Kedro. If anything fails, we'll know then and there.

## Development notes
I've only touched the `.circleci/config.yml` file. The curl command works when run via the CLI, so it should work here, pending I setup things correctly.

Further, a `CIRCLE_TOKEN` environment variable was added to the [project](https://app.circleci.com/settings/project/github/quantumblacklabs/kedro/environment-variables?return-to=https%3A%2F%2Fapp.circleci.com%2Fpipelines%2Fgithub%2Fquantumblacklabs%2Fkedro) for this to work.

Documentation context: I referenced the CircleCI [API docs](https://circleci.com/docs/api/v2/#operation/triggerPipeline) as well as this [support page](https://support.circleci.com/hc/en-us/articles/360047039292-Step-that-triggers-pipeline-in-different-project), which is exactly what was needed.

## Checklist

- [x] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md) file
- [ ] Added tests to cover my changes

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -122,6 +122,18 @@ commands:
...
+    steps:
+      - run:
+          name: Trigger a kedro-viz build
+          command: |
+            curl --location --request POST \
+              --url https://circleci.com/api/v2/project/github/quantumblacklabs/kedro-viz/pipeline \
+              --header "Circle-Token: $CIRCLE_TOKEN" \
+              --header 'content-type: application/json' \
+              --data '{"branch":"main"}'
+
```
Comment (datajoely) [inline .circleci/config.yml:136]: Clever!

**Review by AntonyMilneQB — APPROVED** [review]
I like this 👍  Just to check, since this is added to `workflows` but not `all_circleci_checks_succeeded` we don't actually need the kedro-viz job to return anything or pass in order for us to merge to master here?

File: .circleci/config.yml [hunk 2]
```diff
@@ -122,6 +122,18 @@ commands:
           name: Pip-compile requirements file
           command: make pip-compile
 
+  deploy_viz:
```
Comment (MerelTheisenQB) [inline .circleci/config.yml:125]: This isn't really deploying viz though is it? I think `run_viz_build` might be a more suitable name.

**Review by MerelTheisenQB — APPROVED** [review]
Nice solution 👍

File: .circleci/config.yml (hunk 2 above)
Comment (tynandebold) [reply .circleci/config.yml:125]: Good call. Will change.

**Pull request merged**